### PR TITLE
Fix compatibility renderer load of compressed layered `GL_TEXTURE_2D_ARRAY`

### LINKED ISSUE #91365: VRAM Compressed Texture2DArray formats do not work in Compatibility renderer
### Tested versions

- Reproducible in 4.2.2, 4.3.dev5

### System information

Godot v4.3.dev5 - Windows 10.0.19045 - GLES3 (Compatibility) - NVIDIA GeForce GTX 1060 6GB (NVIDIA; 31.0.15.3713) - Intel(R) Core(TM) i5-3470 CPU @ 3.20GHz (4 Threads)

### Issue description

In the compatibility renderer, textures imported as Texture2DArray seem only work if imported as: lossless, lossy, VRAM uncompressed.
However, if imported as VRAM Compressed or Basis Universal, they display as black both in the inspector as well as when used in scene materials.

This issue does no occur with the Forward+ or Mobile renderers.

image

### Steps to reproduce

Open repro project, check main scene

### Minimal reproduction project (MRP)

[CompatTextArrayTest.zip](https://github.com/godotengine/godot/files/15166552/CompatTextArrayTest.zip)

### PR DESCRIPTION
**Pull request by semensanyok**
# fixes https://github.com/godotengine/godot/issues/91365

### CODE REVIEW

File: drivers/gles3/storage/texture_storage.cpp [hunk 1]
```diff
@@ -1495,14 +1495,17 @@ void TextureStorage::_texture_set_data(RID p_texture, const Ref<Image> &p_image,
 	for (int i = 0; i < mipmaps; i++) {
 		int size, ofs;
 		img->get_mipmap_offset_and_size(i, ofs, size);
-
 		if (compressed) {
-			glPixelStorei(GL_UNPACK_ALIGNMENT, 4);
```
Comment (clayjohn) [inline drivers/gles3/storage/texture_storage.cpp:1500]: This line should not have been deleted. We should be using an alignment of 4 for compressed textures

**Comment by clayjohn** [comment]
I left one suggestion. Other than that everything else looks good and I can confirm that this PR works with the MRP!

File: drivers/gles3/storage/texture_storage.cpp (hunk 1 above)
Comment (semensanyok) [reply drivers/gles3/storage/texture_storage.cpp:1500]: 4 is a default, but better to explicitly define, aggree. Deleted at some point and havent returned. sorry. rolled back.

**Review by clayjohn — APPROVED** [review]
Looks great! Thank you for your quick response

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**